service/src/test/java/com/google/auto/service/processor/AutoServiceProcessorTest.java

```diff
@@ -46,4 +48,30 @@ public void autoService() {
             JavaFileObjects.forResource("META-INF/services/test.SomeService"),
             JavaFileObjects.forResource("META-INF/services/test.AnotherService"));
   }
-}
\ No newline at end of file
+
+  @Test
+  public void multiService() {
+    assert_().about(javaSources())
+        .that(Arrays.asList(
+            JavaFileObjects.forResource("test/SomeService.java"),
+            JavaFileObjects.forResource("test/MultiServiceProvider.java"),
+            JavaFileObjects.forResource("test/AnotherService.java")))
+        .processedWith(new AutoServiceProcessor())
+        .compilesWithoutError()
+        .and().generatesFiles(
+            JavaFileObjects.forResource("META-INF/services/test.SomeService"),
+            JavaFileObjects.forResource("META-INF/services/test.AnotherService"));
+  }
+
+  @Test
+  public void badMultiService() {
+    assert_().about(javaSources())
+        .that(Arrays.asList(
+            JavaFileObjects.forResource("test/SomeService.java"),
```

Comment:
You could remove SomeService and AnotherService, and have BadMultiServiceProvider not implement anything.

Reply:
d2d4c4c

Reply:
Nit: maybe rename `BadMultiService` to `NoServices`?

Reply:
e4c030f